frontend/apps/reader/modules/readerdevicestatus.lua

```diff
@@ -0,0 +1,119 @@
+                        width = Screen:getWidth() * 0.6,
+                        value = curr_items,
+                        value_min = 5,
+                        value_max = 90,
+                        value_hold_step = 10,
+                        ok_text = _("Set threshold"),
+                        title_text = _("Low battery threshold"),
+                        callback = function(battery_spin)
+                            G_reader_settings:saveSetting("low_battery_threshold", battery_spin.value)
+                            powerd:setDissmisBatteryStatus(false)
+                        end
+                    }
+                    UIManager:show(battery_spin)
+                end,
+            },
+        },
+    }
+end
+
+function ReaderDeviceStatus:scheduleBatteryLevel()
+    if G_reader_settings:nilOrTrue("battery_alarm") and (powerd:getCapacity() > 0 or powerd:isCharging()) then
+        self.check_low_battery = function()
+            UIManager:scheduleIn(300, self.check_low_battery)
+            local threshold = G_reader_settings:readSetting("low_battery_threshold") or 20
+            local battery_capacity = powerd:getCapacity()
+            if powerd:getDissmisBatteryStatus() ~= true and battery_capacity <= threshold then
+                local low_battery_info
+                low_battery_info = ButtonDialogTitle:new {
+                    modal = true,
+                    title = T(_("The battery is getting low.\n%1% remaining."), battery_capacity),
+                    title_align = "center",
+                    title_face = Font:getFace("infofont"),
+                    dismissable = false,
+                    buttons = {
+                        {
+                            {
+                                text = _("Dismiss"),
+                                callback = function()
+                                    UIManager:close(low_battery_info)
+                                    powerd:setDissmisBatteryStatus(true)
+                                    UIManager:scheduleIn(300, self.check_low_battery)
+                                end,
+                            },
+                        },
+                    }
+                }
+                UIManager:show(low_battery_info)
+                self:unScheduleBatteryLevel()
+            elseif powerd:getDissmisBatteryStatus() and battery_capacity > threshold then
+                powerd:setDissmisBatteryStatus(false)
+            end
+        end
+        self.check_low_battery()
+    end
+end
+
+function ReaderDeviceStatus:unScheduleBatteryLevel()
+    if self.check_low_battery then
+        UIManager:unschedule(self.check_low_battery)
+        powerd:setDissmisBatteryStatus(false)
+    end
+end
+
+function ReaderDeviceStatus:onResume()
+    self:scheduleBatteryLevel()
+end
+
+function ReaderDeviceStatus:onSuspend()
+    if self.check_low_battery then
+        UIManager:unschedule(self.check_low_battery)
+    end
```

Comment:
Or just `self::unScheduleBatteryLevel()` ?

Reply:
No, because we don't want here `powerd:setDissmisBatteryStatus(false)` - when battery is low and we dismiss earlier we don't now every time after suspend + resume info that battery is low.

Reply:
Ok. No need at some places to set `self.check_low_battery = nil` when it has been UIManager:unschedule()'d, so you don't keep unscheduling something that has already been unscheduled (I guess UIManager would just ignore it, but still)

Reply:
I don't think is good idea because `self:unScheduleBatteryLevel()` is in self.check_low_battery function. When I try set `self.check_low_battery = nil` get:
`./luajit: frontend/ui/uimanager.lua:300: assertion failed!`

Reply:
May be it could be made clearer if:
- you define `self.check_low_battery` only once, in :init() (it does not need to be redefined at each suspend/resume)
- you rename ScheduleBatteryLevel and unScheduleBatteryLevel to startBatteryCheck and stopBatteryCheck (because it's self.check_low_battery that reschedules itself)
- Whenever you reschedule, use:
```lua
UIManager:unschedule(self.check_low_battery) -- to unschedule any previous one
UIManager:scheduleIn(300, self.check_low_battery) -- to schedule a new one
```